### PR TITLE
WIP: Use reflex-ghci

### PR DESCRIPTION
**Pull request by ali-abrar**
Use reflex-ghci. Module loading output is now separated from backend/frontend output. 

I have:

  - [x] Based work on latest `develop` branch
  - [ ] Looked for lint in my changes with `hlint .` (lint found in code you did not write can be left alone)
  - [ ] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

**Comment by srid** [comment]
Tested it on my machine (nixos-unstable). Works great!

File: dep/reflex-platform/github.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "owner": "reflex-frp",
   "repo": "reflex-platform",
-  "branch": "develop",
-  "rev": "90351bdb2c99d416ea95fd5b8b474e8f9fab2942",
-  "sha256": "0bbzz4ckdx0ilhhpd3drfbsc0zdb2rnhw2m3nls9villbs3lz3lh"
+  "branch": "aa-reflex-ghci",
```
Comment (3noch) [inline dep/reflex-platform/github.json]: I'll at least wait till this gets in.

Comment (alexfmpe) [reply dep/reflex-platform/github.json]: Seems to be merged https://github.com/reflex-frp/reflex-platform/pull/620/commits

**Comment by ali-abrar** [comment]
I think we should do more thorough testing and probably some code
deduplication before this lands.

File: dep/reflex-platform/github.json (hunk 1 above)
Comment (3noch) [reply dep/reflex-platform/github.json]: Awesome. Just need to update the branch here then.

**Comment by 3noch** [comment]
I'll start using it myself and see how it goes.

**Comment by srid** [comment]
When the backend spits out sufficient lines, reflex-ghci does not automatically scroll to the tail of it. As the user, I have to manually scroll to the bottom: https://asciinema.org/a/Tp6POfTXFXh51eWiqjBFfrlBY

**Comment by ali-abrar** [comment]
> When the backend spits out sufficient lines, reflex-ghci does not automatically scroll to the tail of it. As the user, I have to manually scroll to the bottom: https://asciinema.org/a/Tp6POfTXFXh51eWiqjBFfrlBY

Thanks for the report. I'll look into this.

**Comment by 3noch** [comment]
Might be worth adding a `ghcid-output.txt`-type-thing as well so that people can interact with the warnings outside of the terminal.

**Comment by 3noch** [comment]
Also need to document how to copy from the terminal it seems?

**Comment by ali-abrar** [comment]
> Also need to document how to copy from the terminal it seems?

Shift-click to copy text

**Comment by 3noch** [comment]
> > Also need to document how to copy from the terminal it seems?
> 
> Shift-click to copy text

**Comment by ali-abrar** [comment]
We'll come back to this later.

**Pull request closed without merging**